File: 680-valid-palindrome-ii/680-valid-palindrome-ii.cpp
class Solution {
public:
    
    bool isPalin(string &s, int st, int en){
        
        while(st<=en && s[st]==s[en]){
            
            st++;
            en--;
        }
        if(en<=st) return 1;
        
        return 0;
    }
    
    
    bool validPalindrome(string s) {
        
        int pt1=0, pt2=s.size()-1;   
        
     const int n=s.size();
        
        int cnt=0;
        while(pt1<=pt2){
            
          
            
            if(s[pt1]==s[pt2]){
                pt1++;
                pt2--;
            }else{
                cnt++;
                if(cnt>1)
                    return 0;
              
                return (isPalin(s,pt1+1,pt2) | isPalin(s,pt1,pt2-1));
            }
        }
        return 1;
    }
};

File: 680-valid-palindrome-ii/680-valid-palindrome-ii.java
class Solution {
    
    boolean isPalind(String s, int st, int en){
        
        while(st<=en){
            if(s.charAt(st) == s.charAt(en)){
                st++;
                en--;
            }else
                return false;
        }
        
        return true;
    }
    

    public boolean validPalindrome(String s) {
      int st=0; 
      int en=s.length()-1;
        
        
        while(st<en){
             if(s.charAt(st) == s.charAt(en)){
                st++;
                en--;
             }else{   
                 return isPalind(s, st+1,en) | isPalind(s, st,en-1);
             } 
        }
    return true;
    }
}